pkg/rpc/response/result/invoke.go

```diff
@@ -108,23 +132,46 @@ arrloop:
 	for i := range arr {
 		var data []byte
 		if (r.Stack[i].Type() == stackitem.InteropT) && iterator.IsIterator(r.Stack[i]) {
-			iteratorValues, truncated := iterator.Values(r.Stack[i], r.maxIteratorResultItems)
-			value := make([]json.RawMessage, len(iteratorValues))
-			for j := range iteratorValues {
-				value[j], err = stackitem.ToJSONWithTypes(iteratorValues[j])
+			if sessionsEnabled {
+				if sessionID == "" {
+					sessionID = uuid.NewString()
+				}
+				iteratorID, err := r.registerIterator(sessionID, r.Stack[i], i, r.finalize)
+				if err != nil {
+					r.FaultException += fmt.Sprintf("%server error: failed to register iterator session: %v", faultSep, err)
```

Comment:
Why don't we return an error from `MarshalJSON` in this case? It feels to me that `FaultException` is more about execution exceptions, while here we've reached some server limit.

Reply:
I thought that the user can at least get some data about invocation like notifications and HALT/FAULT state, but you're right, it's useless without the iterator items.